src/main/java/com/microsoft/sqlserver/jdbc/ISQLServerDataSource.java

```diff
@@ -821,6 +821,21 @@
      */
     String getMSIClientId();
 
+    /**
```

Comment:
Aren't you adding 2 new connection properties?

Reply:
not anymore, latest discussion with Jakub we're aligning with ODBC so using keyStorePrincipalId (new) and keyStoreSecret (existing)